algebird-test/src/test/scala/com/twitter/algebird/MaxLaws.scala

```diff
@@ -0,0 +1,53 @@
+package com.twitter.algebird
+
+import com.twitter.algebird.BaseProperties._
+import com.twitter.algebird.scalacheck.arbitrary._
+import com.twitter.algebird.scalacheck.NonEmptyVector
+import org.scalacheck.Arbitrary
+import org.scalacheck.Prop.forAll
+
+class MaxLaws extends CheckProperties {
+  def maxTest[T: Arbitrary: Ordering] =
+    forAll { (l: Max[T], r: Max[T]) =>
+      val realMax = Max(Ordering[T].max(l.get, r.get))
+      l + r == realMax && (l max r) == realMax
+    }
+
+  property("Max.{ +, max } works on ints") { maxTest[Int] }
+
+  property("Max should work on non-monoid types like String") { maxTest[String] }
+
+  property("Max.aggregator returns the maximum item") {
+    forAll { v: NonEmptyVector[Int] =>
+      v.sorted.last == Max.aggregator[Int].apply(v.items)
+    }
+  }
+
+  property("Max[String] is a commutative semigroup") {
+    commutativeSemigroupLaws[Max[String]]
```

Comment:
I think there is a `commutativeMonoid` here too, right? `""` is the `zero`?

`Min[String]` is the one that is just a semigroup because there is no max string.